core/src/main/java/org/apache/iceberg/metrics/TimerResultParser.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.apache.iceberg.metrics.ScanReport.TimerResult;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+class TimerResultParser {
+  private static final String MISSING_FIELD_ERROR_MSG =
+      "Cannot parse timer from '%s': Missing field '%s'";
+
+  static final TimerResult NOOP_TIMER =
+      new TimerResult("undefined", TimeUnit.NANOSECONDS, Duration.ZERO, 0);
+
+  private static final String NAME = "name";
+  private static final String TIME_UNIT = "time-unit";
+  private static final String COUNT = "count";
+  private static final String TOTAL_DURATION_NANOS = "total-duration-nanos";
```

Comment:
Why is there a unit if this is always in nanos? I think this should match the unit, or we should omit the unit.

Reply:
ah I had a TODO to fix this but forgot about it, thanks for spotting this. Renamed the field to `total-duration` and fixed it now